bin/remove_field.py

```diff
@@ -74,51 +71,26 @@ def wait_times_access():
     yield from [5, 5, 5, 5, 5]
 
 
-@retry(
-    retryable_exceptions=[ClientError],
-    wait_time_generator=wait_times_access,
-    module_logger=logger,
-)
-def fix_data_in_s3(fields, s3_crashstorage, crash_data):
-    """Fix data in raw_crash file in S3."""
-    bucket = s3_crashstorage.bucket
-    s3_client = s3_crashstorage.connection
-
+def fix_data_in_raw(fields, raw_crashstorage, crash_data):
+    """Fix data in raw_crash file."""
     crashid = crash_data["crashid"]
 
-    # Iterate through the available keys until we find one we like.
-    paths = build_keys("raw_crash", crashid)
-    path = None
-    for possible_path in paths:
-        try:
-            resp = s3_client.load_file(bucket=bucket, path=possible_path)
-            path = possible_path
-            break
-        except s3_client.KeyNotFound:
-            continue
-
-    if not path:
-        # We're in a weird situation where we know there's a raw crash, but we can't
-        # find it with the key options.
-        raise Exception(f"raw crash not available at any keys: {paths}")
-
-    raw_crash_as_string = resp["Body"].read()
-    data = json.loads(raw_crash_as_string)
+    data = raw_crashstorage.get_raw_crash(crashid)
+
     should_save = False
     for field in fields:
         if field in data:
             del data[field]
             should_save = True
 
     if should_save:
-        s3_client.save_file(
-            bucket=bucket,
-            path=path,
-            data=io.BytesIO(dict_to_str(data).encode("utf-8")),
-        )
-        click.echo("# s3: fixed raw crash")
+        # NOTE(relud): this is not guaranteed to save the modified raw crash in the same
+        # location as the original, but it will still be read instead of the original if
+        # that happens. In practice as of 2024-07-17 it always overwrites the original.
+        raw_crashstorage.save_raw_crash(raw_crash=data, dumps=None, crash_id=crashid)
```

Comment:
I considered exposing a modified `get_raw_crash_and_path` method, but I think this is sufficient and more maintainable.